### PR TITLE
add doc for update channel apiv4

### PR DESCRIPTION
**Pull request by cpanato**
PR for the implementation: https://github.com/mattermost/platform/pull/5624

### CODE REVIEW

File: source/v4/channels.yaml [hunk 1]
```diff
@@ -49,6 +49,53 @@
...
         '403':
           $ref: '#/responses/Forbidden'
+    put:
+      tags:
+        - channels
+      summary: Update a channel
+      description: |
+        Update a channel.
+        ##### Permissions
+        If updating a public channel, `create_public_channel` permission is required. If creating a private channel, `create_private_channel` permission is required.
```
Comment (jwilander) [inline source/v4/channels.yaml]: The permission should be for manage channel, not create channel right?

File: source/v4/channels.yaml [hunk 2]
```diff
@@ -49,6 +49,53 @@
...
+                description: A short description of the purpose of the channel
+              header:
+                type: string
+                description: Markdown-formatted text to display in the header of the channel
+              type:
+                type: string
+                description: "'O' for a public channel, 'P' for a private channel"
+      responses:
+        '200':
+          description: Channel updated successful
```
Comment (jwilander) [inline source/v4/channels.yaml]: Minor but can we make this `Channel update successful`

File: source/v4/channels.yaml [hunk 3]
```diff
@@ -49,6 +49,53 @@
...
         '403':
           $ref: '#/responses/Forbidden'
+    put:
+      tags:
+        - channels
+      summary: Update a channel
+      description: |
+        Update a channel.
+        ##### Permissions
+        If updating a public channel, `manage_public_channel_members` permission is required. If creating a private channel, `manage_private_channel_members` permission is required.
```
Comment (jwilander) [inline source/v4/channels.yaml]: Oh, actually `If creating a private channel` needs to be `If updating a private channel`

Comment (cpanato) [inline source/v4/channels.yaml]: done

**Pull request merged**